packages/@aws-cdk/aws-cloudfront/lib/origin_access_identity.ts

```diff
@@ -0,0 +1,126 @@
+import iam = require('@aws-cdk/aws-iam');
+import cdk = require('@aws-cdk/core');
+import { CfnCloudFrontOriginAccessIdentity } from './cloudfront.generated';
+
+export interface CloudFrontOriginAccessIdentityProps {
+  /**
+   * Any comments you want to include about the origin access identity.
+   *
+   * @default "Allows CloudFront to reach to the bucket"
+   */
+  readonly comment?: string;
+}
+
+export interface ICloudFrontOriginAccessIdentity extends cdk.IResource, iam.IGrantable  {
+  /**
+   * The Origin Access Identity Name
+   */
+  readonly name: string;
+}
+
+export abstract class BaseCloudFrontOriginAccessIdentity extends cdk.Resource {
+  /**
+   * The Origin Access Identity Name (physical id)
+   */
+  public abstract readonly name: string;
+  /**
+   * Derived principal value for bucket access
+   */
+  public abstract readonly grantPrincipal: iam.IPrincipal;
+
+  /**
+   * The Arn to include in S3 bucket policy to allow CloudFront access
+   */
+  protected arn(): string {
+    return cdk.Stack.of(this).formatArn(
+      {
+        service: "iam",
+        region: "", // global
+        account: "cloudfront",
+        resource: "user",
+        resourceName: `CloudFront Origin Access Identity ${this.name}`
+      }
+    );
+  }
+}
+
+export interface CloudFrontOriginAccessIdentityAttributes {
+  /**
+   * The Origin Access Identity Name
+   */
+  readonly name: string;
+}
+
+/**
+ * An origin access identity is a special CloudFront user that you can
+ * associate with Amazon S3 origins, so that you can secure all or just some of
+ * your Amazon S3 content.
+ */
+export class CloudFrontOriginAccessIdentity extends BaseCloudFrontOriginAccessIdentity implements ICloudFrontOriginAccessIdentity {
+  public static fromCloudFrontOriginAccessIdentityAttributes(
+      scope: cdk.Construct,
+      id: string,
+      attrs: CloudFrontOriginAccessIdentityAttributes): BaseCloudFrontOriginAccessIdentity {
+
+    class Import extends BaseCloudFrontOriginAccessIdentity {
+      public readonly name = attrs.name;
+      public readonly grantPrincipal = new iam.ArnPrincipal(this.arn());
+    }
+
+    return new Import(scope, id);
+  }
+
+  public static fromName(
+    scope: cdk.Construct,
+    id: string,
+    name: string): BaseCloudFrontOriginAccessIdentity {
+    return CloudFrontOriginAccessIdentity.fromCloudFrontOriginAccessIdentityAttributes(scope, id, {
+      name
+    });
+  }
+
+  /**
+   * CDK L1 resource
+   */
+  public readonly resource: CfnCloudFrontOriginAccessIdentity;
+
+  /**
+   * The Amazon S3 canonical user ID for the origin access identity, used when
+   * giving the origin access identity read permission to an object in Amazon
+   * S3.
+   *
+   * @attribute S3CanonicalUserId
+   */
+  public readonly s3CanonicalUserId: string;
```

Comment:
Attributes must begin with the resource name

Reply:
I was hoping it's not required since the name is super long `this.cloudFrontOriginAccessIdentityS3CanonicalUserId` - but sure, changed it.

Does that apply to `Name` as well? https://github.com/aws/aws-cdk/pull/4491/files#r334299183